test/integration/stats_integration_test.cc

```diff
@@ -337,5 +429,62 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeHostSizeWithStats) {
   EXPECT_MEMORY_LE(m_per_host, 2000); // Round up to allow platform variations.
 }
 
+class EdsClusterMemoryTestRunner : public testing::TestWithParam<Network::Address::IpVersion> {
+protected:
+  EdsClusterMemoryTestRunner()
+      : ip_version_(testing::TestWithParam<Network::Address::IpVersion>::GetParam()) {}
+
+  Network::Address::IpVersion ip_version_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, EdsClusterMemoryTestRunner,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+TEST_P(EdsClusterMemoryTestRunner, MemoryLargeHostSizeWithStats) {
+  // A unique instance of ClusterMemoryTest allows for multiple runs of Envoy with
+  // differing configuration. This is necessary for measuring the memory consumption
+  // between the different instances within the same test.
+  const size_t m_delta = ClusterMemoryTestHelper::computeEdsMemoryDelta(1, 1, 1, 101, 1, 1, true);
+  FANCY_LOG(debug, "delta memory = {}", m_delta);
+
+  // We only run the exact test for ipv6 because ipv4 in some cases may allocate a
+  // different number of bytes. We still run the approximate test.
+  if (ip_version_ != Network::Address::IpVersion::v6) {
+    // https://github.com/envoyproxy/envoy/issues/12209
+    // EXPECT_MEMORY_EQ(m_per_host, 1380);
```

Comment:
the exact compares are all commented out to avoid test-failures due to some relatively new source of non-determinism. But there are still EXPECT_MEMORY_LE tests, right?

Do you want to add some of those here?

Reply:
That is the ultimate goal but I am there yet. 
I hitting some cases that I haven't pinpointed yet. The initial 1 cluster 1 host use more heap than 1 cluster 101 hosts :(
That's why I cannot enable approximate test (EXPECT_LE).

EXPECT Memory EQ is not realistic. I cannot drill down to a single factor yet

```
[2021-03-01 03:28:29.030][12][warning][test/integration/stats_integration_test.cc] [test/integration/stats_integration_test.cc:182] initial memory for 1 cluster 1 hosts is 2905336
[2021-03-01 03:28:34.225][12][warning][test/integration/stats_integration_test.cc] [test/integration/stats_integration_test.cc:189] final memory for 1 cluster 101 hosts is 873344
[2021-03-01 03:28:34.263][12][warning][test/integration/stats_integration_test.cc] [test/integration/stats_integration_test.cc:453] delta memory = 18446744073707519624
```

Reply:
Yeah some bit of non-determinism that wasn't there 6 months ago is now in Envoy and makes EQ memory tests, at least at a test of this scale, not viable. I think it would be great to find and squash the non-determinism, but no one can really prioritize that. It might get fixed silently at some point as it likely affects something else as well.

However the fact that your LE tests are not working as expected suggests a math error somewhere I suspect :)